PB-IDE-Tools/displayCcode.pb

```diff
@@ -0,0 +1,340 @@
+If program
+  While ProgramRunning(program)
+    While AvailableProgramOutput(program)
+      compilerVersion$ + ReadProgramString(program) + #CRLF$
+    Wend
+  Wend
+  isCompilerError = Bool(ProgramExitCode(program))
+  CloseProgram(program)
+Else
+  isCompilerError = #True
+EndIf
+
+If FindString(compilerVersion$, "PureBasic 6.")
+  isCompilerV6 = #True
+EndIf
+
+compilerHomePath$ = GetPathPart(compilerFilePath$)
+codeFilePath$     = ProgramParameter(0) ; "%FILE"
+codeTempFilePath$ = ProgramParameter(1) ; "%TEMPFILE"
+
+If codeFilePath$ = ""
+  ; The code has not yet been saved
+  ; Use the path of the auto-generated temp code file
+  codeFilePath$ = codeTempFilePath$
+EndIf
+
+CompilerIf #PB_Compiler_OS = #PB_OS_Windows
+  asmCodeFilePath$ = compilerHomePath$ + "purebasic.c"
+CompilerElse
+  asmCodeFilePath$ = GetPathPart(codeFilePath$) + "purebasic.c"
+CompilerEndIf
+
+; The EXE file is only created so that the PB compiler does not execute the code, but only creates the ASM code
+exeFilePath$ = GetTemporaryDirectory() + "purebasic"
+
+CompilerIf #PB_Compiler_OS = #PB_OS_Windows
+  workingDirectoryPath$ = GetPathPart(compilerFilePath$)
+CompilerElse
+  workingDirectoryPath$ = GetPathPart(codeFilePath$)
+CompilerEndIf
+
+compilerParameters$ = "--commented"
+
+; Parameter for constants change since pb v6.00
+If isCompilerV6
+  compilerParameters$ + " -co PB_Editor_BuildCount=0 -co PB_Editor_CompileCount=0"
+Else
+  compilerParameters$ + " -o PB_Editor_BuildCount=0 -o PB_Editor_CompileCount=0"
+EndIf
+
+If Val(GetEnvironmentVariable("PB_TOOL_Thread"))
+  compilerParameters$ + " --thread"
+EndIf
+
+If Val(GetEnvironmentVariable("PB_TOOL_Unicode"))
+  compilerParameters$ + " --unicode"
+EndIf
+
+If GetEnvironmentVariable("PB_TOOL_SubSystem")
+  compilerParameters$ + " --subsystem " + GetEnvironmentVariable("PB_TOOL_SubSystem")
+EndIf
+
+Procedure$ ProcessParameter(parameter$, exeFilePath$, *isLibrary.Integer)
+  Select LCase(parameter$)
+    Case "-dl", "--dylib", "-so", "--sharedobject"
+      parameter$ = parameter$ + " " + #DQUOTE$ + exeFilePath$ + #DQUOTE$
+      *isLibrary\i = #True
+    Case "/dll"
+      parameter$ = "--executable " + #DQUOTE$ + exeFilePath$ + #DQUOTE$ + parameter$
+      *isLibrary\i = #True
+  EndSelect
+  ProcedureReturn parameter$
+EndProcedure
+
+; Get user parameters
+countOfParameters = CountProgramParameters()
+For i = 2 To countOfParameters - 1
+  compilerUserParameters$ + " " + ProcessParameter(ProgramParameter(i), exeFilePath$, @isLibrary)
+Next
+If Not isLibrary
+  compilerParameters$ + " --executable " + #DQUOTE$ + exeFilePath$ + #DQUOTE$
+EndIf
+compilerParameters$ + compilerUserParameters$
+
+; =======================
+;-Delete Old Output Files
+; =======================
+
+; Make sure that there are no previous output files, to prevent old output files from being read, if an error occurs.
+DeleteFile(asmCodeFilePath$)
+DeleteFile(exeFilePath$)
+
+; ====================
+;-Create ASM Code File
```

Comment:
Please replace `ASM` with `C`. There are more places in which this should be done, too.